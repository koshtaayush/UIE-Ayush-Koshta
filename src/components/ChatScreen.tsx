import React from 'react';
import styled from 'styled-components';
import bodyBackground from './../assets/bodyBackground.png';
import SendMessage from './SendMessage';
import MessageList from './MessageList';
import { IMessage } from './../typings/sharedInterface';

interface Props {}

const ChatScreen: React.FC<Props> = () => {
    
    // The message sent by the current user gets temporarily stored in the local state 
    //    (which is just being used to render the message) to avoid excess fetch API call
    const [myAdditionalMessages, setMyAdditionalMessages] = React.useState<Array<IMessage>>([]);

    /**
     * Function to add the message sent by user to temporary list
     */
    const addCurrentMessages = (currentMessage: IMessage) => {
        setMyAdditionalMessages([...myAdditionalMessages, ...[currentMessage]])
    }

    /**
     * Function to clear the temporary message list after the API call
     */
    const nullifyAdditionalMessage = (): void => {
        setMyAdditionalMessages([]);
    }

    return (
        <React.Fragment>
            <ChatApplicationSection>
                <ChatApplication>
                    <Container>
                        <MessageList 
                            myAdditionalMessages={myAdditionalMessages} 
                            nullifyAdditionalMessage={nullifyAdditionalMessage} />
                        <SendMessage 
                            addCurrentMessages={addCurrentMessages} />
                    </Container>
                </ChatApplication>
            </ChatApplicationSection>
        </React.Fragment>
    )


}

export default ChatScreen

const ChatApplicationSection = styled.div`
    height: 100%;
    display: flex;
    justify-content: center;
    background: url(${bodyBackground});
`

const ChatApplication = styled.div`
    display: flex;
    justify-content: flex-end;
    flex-direction: column;
    align-items: center;
    height: calc(100vh - 7.5rem);
    width: 100%;
    max-width: 40rem;
`

const Container  = styled.div`
    height: 100%;
    width: 100%;
    max-width: 40rem;
`