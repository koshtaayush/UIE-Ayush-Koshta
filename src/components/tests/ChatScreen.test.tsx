import React from 'react'
import { shallow } from 'enzyme'

import ChatScreen from './../ChatScreen';

describe('ChatScreen', () => {

    it('Should match the snapshot', () => {

        const component = shallow(<ChatScreen />);  
        expect(component).toMatchSnapshot();
    })

    it('Should always render Send Message Bar', () => {

        const component = shallow(<ChatScreen />);  
        expect(component.find({ 'test-id': 'sendMessageBar' })).toHaveLength(1);
    })

})
