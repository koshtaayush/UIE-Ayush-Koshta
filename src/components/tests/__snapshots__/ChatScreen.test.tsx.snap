// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen Should match the snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <MessageList
          myAdditionalMessages={Array []}
          nullifyAdditionalMessage={[Function]}
        />
        <SendMessage
          addCurrentMessages={[Function]}
          test-id="sendMessageBar"
        />
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
